app/Http/Controllers/AdminController.php

```diff
@@ -248,47 +248,29 @@ public function removeBuilds(): View|RedirectResponse
             $yearTo = intval($_POST['yearTo']);
         }
 
-        $xml = '<cdash>';
-        $xml .= '<cssfile>' . $config->get('CDASH_CSS_FILE') . '</cssfile>';
-        $xml .= '<version>' . $config->get('CDASH_VERSION') . '</version>';
-        $xml .= '<menutitle>CDash</menutitle>';
-        $xml .= '<menusubtitle>Remove Builds</menusubtitle>';
-        $xml .= '<backurl>manageBackup.php</backurl>';
-
         // List the available projects
-        $projects = $db->executePrepared('SELECT id, name FROM project');
+        $available_projects = [];
+        $projects = DB::select('SELECT id, name FROM project');
         foreach ($projects as $projects_array) {
-            $xml .= '<availableproject>';
-            $xml .= add_XML_value('id', $projects_array['id']);
-            $xml .= add_XML_value('name', $projects_array['name']);
-            if ($projects_array['id'] == $projectid) {
-                $xml .= add_XML_value('selected', '1');
-            }
-            $xml .= '</availableproject>';
+            $available_project = new Project();
+            $available_project->Id = (int) $projects_array->id;
+            $available_project->Name = $projects_array->name;
+            $available_projects[] = $available_project;
         }
 
-        $xml .= '<dayFrom>' . $dayFrom . '</dayFrom>';
-        $xml .= '<monthFrom>' . $monthFrom . '</monthFrom>';
-        $xml .= '<yearFrom>' . $yearFrom . '</yearFrom>';
-        $xml .= '<dayTo>' . $dayTo . '</dayTo>';
-        $xml .= '<monthTo>' . $monthTo . '</monthTo>';
-        $xml .= '<yearTo>' . $yearTo . '</yearTo>';
-
-        @$submit = $_POST['Submit'];
-
         // Delete the builds
-        if (isset($submit)) {
-            $build = $db->executePrepared("
+        if (isset($_POST['Submit'])) {
+            $build = DB::select("
                          SELECT id
                          FROM build
                          WHERE
-                             projectid=?
+                             projectid = ?
                              AND parentid IN (0, -1)
-                             AND starttime<=?||'-'||?||'-'||?||' 00:00:00'
-                             AND starttime>=?||'-'||?||'-'||?||' 00:00:00'
+                             AND starttime <= CAST(CONCAT(?, '-', ?, '-', ?, ' 00:00:00') AS timestamp)
```

Comment:
I get a 500 error when trying to run the remove builds for a couple of days time:

![image](https://github.com/Kitware/CDash/assets/542106/29218cbe-2269-45df-ac0e-9165886991c8)


Is this supposed to be `DATETIME`?

Reply:
This works in Postgres, but not in MySQL.  I'll push a fix.